include/Beman/Execution26/detail/sender_decompose.hpp

```diff
@@ -0,0 +1,137 @@
+// include/Beman/Execution26/detail/sender_decompose.hpp              -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_SENDER_DECOMPOSE
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_SENDER_DECOMPOSE
+
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+// ----------------------------------------------------------------------------
+
+namespace Beman::Execution26::Detail
+{
+    inline constexpr struct sender_any_t
+    {
+        template <typename T> constexpr operator T() const;
+    } sender_any;
+
+    template <typename Tag, typename Data, typename Children>
+    struct sender_meta
+    {
+        using tag_type = ::std::remove_cvref_t<Tag>;
+        using data_type = ::std::remove_cvref_t<Data>;
+        using children_type = ::std::remove_cvref_t<Children>;
+    };
+
+    template <typename Tag, typename Data, typename Children>
+    struct sender_data
+    {
+        ::std::remove_cvref_t<Tag> tag;
+        Data&                      data;
+        Children                   children;
+    };
+
+    template <typename Sender>
+    auto get_sender_meta(Sender&& sender)
+    {
+        #if 0
+        //-dk:TODO should use a dynamic/language approach:
+        auto&& [tag, data, ... children] = sender;
+        return sender_meta<decltype(tag), decltype(data), ::std::tuple<decltype(children)...>>;
+        #endif
+        using sender_type = ::std::remove_cvref_t<Sender>;
+        static constexpr ::Beman::Execution26::Detail::sender_any_t any{};
```

Comment:
obviously we're in a different namespace here, but would we want to avoid the name any -- just to not confuse with std::any?  Or maybe this is actually just a temporary shim?